### ISSUE TITLE
Hit and Watchlist sorts are broken on Flights page

### ISSUE THREAD

**Issue by copenhafer** [US-CBP/GTAS#364]
Did not have time to investigate.  Does not appear to be purely a front-end issue.  Tried reverting to old column headings and problem persisted.

**Comment by tjwdev** [comment]
Took me a moment, but I think I figured it out. The crux of the issue can be found in the commit "Fix flight hit counts". The problem traces back to fixing flight hit counts, particularly their accuracy. 

Our targetingServiceImpl used to have a section in it that updated the fight hit count depending on when it found a hit. This had apparently chances for collisions and what ended up happening was an inaccurate display of hits per flight when searched. That section was removed and the hits counts were then moved to just before the value object was returned AFTER the query was run, calculating then and guaranteeing accuracy. 

Here's the problem: The old targettingServiceImpl section actually updated the flight record table WITH those hit values. It no longer does because of inaccuracies. If you take a look at the flight table on dev you will find every hit count with a 0 for its value. This would not ordinarily be a problem, however, the typed query that pulls back based on a particular column pulls from that flight table. So when you select watch list hits, or rule list hits as the sorting column it checks that tables, sees all 0s, returns the appropriate page and number of values because ALL of those flights based on the sorting column are equivalent, hands it back to the flightServiceImpl where THEN the rule/watchlist hits are counted and added to the value object so that the display is accurate. 

So it goes Sort Request On Hits To Request Mapping-> Backend checks database, query specifically looks at flight table for hits values -> Sees 0s in all hits -> Sorts all 0's descending/ascending (i.e. no movement) -> returns to request mapping method, in that method we do real count of hits for accuracy, adds them to value object -> returns value object which consists of a list of flights of however many display items sorted as if all flights had 0 hits. Which is to say, not sorted.

This results in you seeing all of the flights on the front-end WITH rule hits but you cannot sort, because as far as the backend is concerned, all the flights have 0 hits, so when you sort ascending/descending you're sorting against all 0s and it basically won't do anything. Ta-da!

Few different ways to solve it, I do NOT recommend having the front-end control pagination or sorting for flights. Too much data. Bad idea. Just need to keep that table updated appropriately BEFORE sorting.
